src/main/java/kazoo/service/PartituraService.java

```diff
@@ -38,21 +39,32 @@ public void crearPartitura(String usuarioNombre, Partitura partitura) {
     }
 
     public DetallePartitura getPartitura(String nombreUsuario, String partituraId) {
-        Usuario usuario = getUsuario(nombreUsuario);
-        Partitura partitura = partituraRepository.findById(Long.parseLong(partituraId))
-                .orElseThrow(() -> new PartituraNoEncontradaException("No existe la partitura seleccionada"));
-        validarPartitura(usuario,partitura);
+
+        Partitura partitura = validarPartitura(nombreUsuario,Long.parseLong(partituraId));
         return new DetallePartitura(partitura);
     }
 
-    private void validarPartitura(Usuario usuario, Partitura partitura) {
+    private Partitura validarPartitura(String nombreUsuario, Long id) {
+        Partitura partitura = partituraRepository.findById(id)
+                .orElseThrow(() -> new PartituraNoEncontradaException("No existe la partitura seleccionada"));
+        Usuario usuario = getUsuario(nombreUsuario);
+
         if(!partitura.getUsuario().equals(usuario)){
             throw new PartituraNoEncontradaException("No existe la partitura par el usuario solicitado");
         }
+
+        return partitura;
     }
 
     private Usuario getUsuario(String nombreUsuario) {
         return usuarioRepository.findByNombre(nombreUsuario)
                 .orElseThrow(() -> new DatosDeLogueoInvalidosException("No existe el usuario"));
     }
+
+    public void guardarPartitura(String nombreUsuario, Partitura partituraRecibida) {
+        Partitura partitura = validarPartitura(nombreUsuario, partituraRecibida.getPartitura_id());
+        //Copia todas las propiedades modificadas de la partitura menos el usuario que llega con null
```

Comment:
No entendi, de donde llega el usuario con null? Y por que? :eyes:

Reply:
para pegarle este endpoint, la partitura que usa el front es la que le llega a partir del get, que no devuelve la partitura completa con los datos del usuario por un tema de seguridad (el usuario completo te devolvería el salt y la pss, lo cual es  inaceptable aunque hagamos el tp mas bizarro del mundo). Lo que hicimos en el get de partitura es que devuelva un objeto estilo TO que tiene el detalle de una partitura, obviando la información de usuario (solo tenemos el nombre de usuario que es PK) . Con eso modificado le pega al put, y le manda el nombre de usuario por header para que valide que el usuario con ese nombre, tenga esa partitura.

Reply:
Piola, igual tiene sentido que si pido una partitura no venga el usuario. Ahora entiendo mejor, me sonaba raro `el usuario llega en null` cuando uno de los parametros es nombreUsuario jaja 👍